spec/controllers/marks_graders_controller_spec.rb

```diff
@@ -128,4 +128,253 @@
       expect(response.media_type).to eq 'text/csv'
     end
   end
+
+  describe '#assign_all' do
+    before do
+      allow(GradeEntryStudent).to receive(:assign_all_tas)
+    end
+
+    context 'when students and graders are selected' do
+      it 'calls GradeEntryStudent `assign_all` and return a success response' do
+        student = create(:student)
+        grade_entry_student = grade_entry_form.grade_entry_students.find_or_create_by(role: student)
+        grade_entry_student_ta = create(:grade_entry_student_ta, grade_entry_student: grade_entry_student)
+
+        expect(GradeEntryStudent).to receive(:assign_all_tas).with(
+          [grade_entry_student.id.to_s],
+          [grade_entry_student_ta.id.to_s],
+          grade_entry_form
+        )
+
+        post_as instructor, :assign_all, params: {
+          course_id: course.id,
+          grade_entry_form_id: grade_entry_form.id,
+          students: [grade_entry_student.id],
+          graders: [grade_entry_student_ta.id]
+        }
+
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'when students are not selected' do
+      it 'returns bad request and sets flash error' do
+        post_as instructor, :assign_all, params: {
+          course_id: course.id,
+          grade_entry_form_id: grade_entry_form.id,
+          students: [],
+          graders: [1]
+        }
+
+        expect(response).to have_http_status(:bad_request)
+        expect(flash[:error]).to eq(['<p>Select a student</p>'])
```

Comment:
Use the internationalized strings when checking the flash messages (by calling `I18n.t`)